src/app/utils/query-parameter-sanitization.ts

```diff
@@ -0,0 +1,396 @@
+// OData filter operators
+const COMPARISON_OPERATORS = ['eq', 'ne', 'gt', 'ge', 'lt', 'le'];
+const LOGICAL_OPERATORS = ['and', 'or', 'not'];
+const COLLECTION_OPERATORS = ['any', 'all'];
+const QUERY_FUNCTIONS = ['startswith', 'endswith', 'contains', 'substring', 'indexof', 'concat'];
+
+// REGEXES
+const ALL_ALPHA_REGEX = /^[a-z]+$/i;
+const POSITIVE_INTEGER_REGEX = /^[1-9]+$/;
+// Matches the format json or application/json
+const MEDIA_TYPE_REGEX = /^[a-z]+\/{0,1}[a-z]+$/i;
```

Comment:
I'm not really validating the types but their formats. They could be anything e.g. `application/json;odata=verbose` or `xml` or an invalid type like `media/type` as long as it's all text separated by `/` or  text that is all letters.

Reply:
Because it only matches characters on both sides of `/`, sanitization process will mark `application/octet-stream` as `<invalid-media-type>`. Is that desired?

For full list of media types, please refer to http://www.iana.org/assignments/media-types/media-types.xhtml, which contains non-alpha characters as well. I don't know which subset of those are allowed in the formats you mention though.

Reply:
Thanks for sharing the list. I'll use it to update the regex. Much  appreciated.

Reply:
I updated the regex

Reply:
We also need to support image/jpeg

GET https://graph.microsoft.com/v1.0/me/photo/$value

Reply:
We don't want to miss out on this image: 
![image](https://user-images.githubusercontent.com/8527305/99592193-b9972d80-29a4-11eb-9fd9-7049549e77c5.png)

Reply:
Michael, I realized the comment was misleading. I've updated it.